data/transactions/logic/eval.go

```diff
@@ -3154,3 +3199,124 @@ func opAppParamsGet(cx *evalContext) {
 	cx.stack[last] = value
 	cx.stack = append(cx.stack, stackValue{Uint: exist})
 }
+
+func authorizedSender(cx *evalContext, addr basics.Address) bool {
+	appAddr, err := cx.getApplicationAddress()
+	return err == nil && bytes.Equal(addr[:], appAddr)
+	// This should also accept accounts that have been rekeyed to the appAddr
```

Comment:
Apps will also have authority over accounts that are rekeyed to their address.  This is both more consistent, and will allow some interesting tricks to "hand over" accounts to the app which can be used to store more state (in the account's local state for the app that now controls it!)

Reply:
not sure I understand. If I own account A and applicationAddress is P, then I can rekey A to P once. I do not see how  would P get more space from A since they are physically different.

Reply:
A would also have to be opted into the app.  If it is, then P can store into its local state.  This state is more like global state (because nobody can take it away).  Normally, the extra state that an app gets when an address opts in is subject to ClearState at any time.

I suppose this is mostly already true. You could opt a fresh account into an account, and then rekey it to zero, so nobody could remove the local state.  So I guess the only real new thing here is that it effectively gives the app a separate account to store algos / assets into and spend from them.